analytics/client.py

```diff
@@ -34,6 +35,12 @@ def __init__(self, write_key=None, debug=False, max_queue_size=10000,
         self.debug = debug
         self.send = send
 
+        # On program exit, allow the consumer thread to exit cleanly.
+        # This prevents exceptions and a messy shutdown when the interpreter is
+        # destroyed before the daemon thread finishes execution. However, it
+        # is *not* the same as flushing the queue! To guarantee all
+        atexit.register(self.join)
```

Comment:
derp, will complete the comment. Thanks for catching that... embarrassing.

I thought about making it a non-daemon thread, but that means it won't prevent the program from exiting. This isn't the behavior we want, since the upload thread shouldn't affect the main program's behavior.